src/ui-schema.ts

```diff
@@ -150,13 +152,18 @@ export type OneOfControlOptions = {
 export type AnyOfControlOptions = OneOfControlOptions
 
 export type TextControlType = "multiline" | "password" | "singleline"
+export type TextControlInputProps =
+  | (InputProps & { type: "singleline" })
+  | (TextAreaProps & { type: "multiline" })
+  | (InputProps & { type: "password" })
 
 export type TextControlOptions = {
   type?: TextControlType
   tooltip?: string
   placeholderText?: string
   required?: boolean
   rules?: AntDRule[]
+  inputProps?: TextControlInputProps
```

Comment:
This would require users to specify type in two places. If you add the appropriate inputProps type on lines 156 - 158 instead, this isn't a problem